### PR TITLE
Update otel-proto-go to v0.15.0

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
## Which problem is this PR solving?
Updates otel-proto-go package to latest which includes swapping InstrumenationSpans for ScopeSpans. This is a direct replacement, with the InstrumentationSpans field being deprecated and will be removed in the future.

This change also removes the previously deprecated status_code_ok.

## Short description of the changes
- update otel-proto-go to v0.15.0, includes running `go mod tidy`
- remove deprecated status_code_ok
- support both scope spans and instrumentation spans, capturing the library/scope and adding to resource attrs
- update tests to use the newer scopespans and add test that explicitly uses instrumentation spans

### CODE REVIEW

**Review by JamieDanielson — COMMENTED** [review]
This looks great! Per our team's conversation earlier, let's double check how this impacts hound. Also, not sure where we left off on whether to use `scope.name` instead of `library.name` but want to mention that as a point of consideration as well, at least for the new `ScopeSpans`.

File: otlp/trace.go [hunk 1]
```diff
@@ -90,103 +90,29 @@ func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest, ri
...
-						"meta.annotation_type": "span_event",
-					}
-
-					if sevent.Attributes != nil {
-						addAttributesToMap(attrs, sevent.Attributes)
+		if len(resourceSpan.ScopeSpans) > 0 {
+			for _, scopeSpan := range resourceSpan.ScopeSpans {
+				if scopeSpan.Scope != nil {
+					if len(scopeSpan.Scope.Name) > 0 {
+						resourceAttrs["library.name"] = scopeSpan.Scope.Name
```
Comment (robbkidd) [inline otlp/trace.go:97]:
```suggestion
						resourceAttrs["library.name"] = scopeSpan.Scope.Name
						resourceAttrs["scope.name"] = scopeSpan.Scope.Name

```

File: otlp/trace.go [hunk 2]
```diff
@@ -90,103 +90,29 @@ func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest, ri
...
 					}
-					for k, v := range resourceAttrs {
-						attrs[k] = v
+					if len(scopeSpan.Scope.Version) > 0 {
+						resourceAttrs["library.version"] = scopeSpan.Scope.Version
```
Comment (robbkidd) [inline otlp/trace.go:100]:
```suggestion
						resourceAttrs["library.version"] = scopeSpan.Scope.Version
						resourceAttrs["scope.version"] = scopeSpan.Scope.Version
```

File: otlp/trace.go [hunk 3]
```diff
@@ -90,103 +90,29 @@ func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest, ri
...
-
-					if slink.Attributes != nil {
-						addAttributesToMap(attrs, slink.Attributes)
+				events = toEvents(scopeSpan.Spans, resourceAttrs)
+			}
+		} else {
+			for _, librarySpan := range resourceSpan.InstrumentationLibrarySpans {
+				if librarySpan.InstrumentationLibrary != nil {
+					if len(librarySpan.InstrumentationLibrary.Name) > 0 {
+						resourceAttrs["library.name"] = librarySpan.InstrumentationLibrary.Name
```
Comment (robbkidd) [inline otlp/trace.go:109]:
```suggestion
						resourceAttrs["library.name"] = librarySpan.InstrumentationLibrary.Name
						resourceAttrs["scope.name"] = librarySpan.InstrumentationLibrary.Name
```

File: otlp/trace.go [hunk 4]
```diff
@@ -90,103 +90,29 @@ func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest, ri
...
 					}
-					for k, v := range resourceAttrs {
-						attrs[k] = v
+					if len(librarySpan.InstrumentationLibrary.Version) > 0 {
+						resourceAttrs["library.version"] = librarySpan.InstrumentationLibrary.Version
```
Comment (robbkidd) [inline otlp/trace.go:112]:
```suggestion
						resourceAttrs["library.version"] = librarySpan.InstrumentationLibrary.Version
						resourceAttrs["scope.version"] = librarySpan.InstrumentationLibrary.Version
```

**Review by robbkidd — COMMENTED** [review]
I propose that we double-enter library/scope fields with [the spec declaring Scope as the instrumentation origin](https://github.com/open-telemetry/opentelemetry-specification/pull/2276/files) and [the proto being updated](https://github.com/open-telemetry/opentelemetry-proto/pull/362/files) to _deprecate_ "library" with library intended for removal in June of this year.

Maybe we keep duplicating these fields for the foreseeable future.
Maybe we announce that we will stop setting `library.*` after June 15.
Maybe we only stop setting `library.*` for `Scope` and keep setting `library.*` for folks sending us OTel from old instrumentations.

Double-entry gives us options, I think.

**Comment by MikeGoldsmith** [comment]
Yeah, I agree - we can double add the attr and review in future. I've added this with tests 👍🏻

**Comment by MikeGoldsmith** [comment]
Marking as blocked while we review otlp proto versions and use of deprecated fields which have been removed in v0.15.

**Comment by JamieDanielson** [comment]
This closes #64 right?

**Comment by MikeGoldsmith** [comment]
I thought we confirmed only certain combinations of otel-go sdk with updated otel-proto-go packages? If so, then we can close #64 now and leave this open until we upgrade to 0.15+.

**Comment by MikeGoldsmith** [comment]
Moving to icebox for now as we can't upgrade yet.

**Comment by MikeGoldsmith** [comment]
Closing as we'll move up latest proto spec version, currently v0.17

**Pull request closed without merging**